File: src/main/java/work/xeltica/craft/core/stores/QuickChatStore.java
package work.xeltica.craft.core.stores;

import org.bukkit.entity.Player;
import work.xeltica.craft.core.XCorePlugin;
import work.xeltica.craft.core.api.Ticks;

import java.io.IOException;
import java.util.Set;

/**
 * @author raink1208
 */
public class QuickChatStore {
    public QuickChatStore() {
        instance = this;
        XCorePlugin.getInstance().saveResource("quickChats.yml", false);
        config = new Ticks.Config("quickChats");
    }

    public static QuickChatStore getInstance() { return instance; }

    public Set<String> getAllPrefix() {
        return config.getConf().getKeys(false);
    }

    public String getMessage(String prefix) {
        return config.getConf().getString(prefix);
    }

    public String chatFormat(String msg, Player player) {

        msg = msg.replace("{world}", WorldStore.getInstance().getWorldDisplayName(player.getWorld()));
        msg = msg.replace("{x}", String.valueOf(player.getLocation().getBlockX()));
        msg = msg.replace("{y}", String.valueOf(player.getLocation().getBlockY()));
        msg = msg.replace("{z}", String.valueOf(player.getLocation().getBlockZ()));

        return msg;
    }

    public boolean register(String prefix, String msg) {
        if (config.getConf().contains(prefix)) return false;

        config.getConf().set(prefix, msg);
        try {
            config.save();
        } catch (IOException e) {
            e.printStackTrace();
        }
        return true;
    }

    public boolean unregister(String prefix) {
        if (config.getConf().contains(prefix)) {
            config.getConf().set(prefix, null);
            try {
                config.save();
            } catch (IOException e) {
                e.printStackTrace();
            }
            return true;
        }
        return false;
    }

    private static QuickChatStore instance;
    private final Ticks.Config config;
}

File: src/main/java/work/xeltica/craft/core/stores/QuickChatStore.kt
package work.xeltica.craft.core.stores

import org.bukkit.entity.Player
import java.io.IOException
import work.xeltica.craft.core.XCorePlugin
import work.xeltica.craft.core.api.Config

/**
 * @author raink1208
 */
class QuickChatStore {
    val allPrefix: Set<String>
        get() = config.conf.getKeys(false)

    fun getMessage(prefix: String?): String {
        return config.conf.getString(prefix!!)!!
    }

    fun chatFormat(msg: String, player: Player): String {
        var msg = msg
        msg = msg.replace("{world}", WorldStore.getInstance().getWorldDisplayName(player.world))
        msg = msg.replace("{x}", player.location.blockX.toString())
        msg = msg.replace("{y}", player.location.blockY.toString())
        msg = msg.replace("{z}", player.location.blockZ.toString())
        return msg
    }

    fun register(prefix: String?, msg: String?): Boolean {
        if (config.conf.contains(prefix!!)) return false
        config.conf[prefix] = msg
        try {
            config.save()
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return true
    }

    fun unregister(prefix: String?): Boolean {
        if (config.conf.contains(prefix!!)) {
            config.conf[prefix] = null
            try {
                config.save()
            } catch (e: IOException) {
                e.printStackTrace()
            }
            return true
        }
        return false
    }

    private val config: Config

    init {
        instance = this
        XCorePlugin.instance.saveResource("quickChats.yml", false)
        config = Config("quickChats")
    }

    companion object {
        @JvmStatic
        lateinit var instance: QuickChatStore
            private set
    }
}